pkg/storage/stores/indexshipper/compactor/table.go

```diff
@@ -127,7 +131,9 @@ func (t *table) compact(applyRetention bool) error {
 	t.usersWithPerUserIndex = usersWithPerUserIndex
 
 	level.Info(t.logger).Log("msg", "listed files", "count", len(indexFiles))
```

Comment:
There's a case above where `len(indexFiles) == 0`, we probably want to collect that metric too?